src/natcap/invest/sdr/sdr.py

```diff
@@ -557,15 +577,31 @@ def execute(args):
         args['biophysical_table_path'],
         **MODEL_SPEC['args']['biophysical_table_path'])
 
-    # Test to see if c or p values are outside of 0..1
-    for key in ['usle_c', 'usle_p']:
-        for lulc_code, row in biophysical_df.iterrows():
-            if row[key] < 0 or row[key] > 1:
-                raise ValueError(
-                    f'A value in the biophysical table is not a number '
-                    f'within range 0..1. The offending value is in '
-                    f'column "{key}", lucode row "{lulc_code}", '
-                    f'and has value "{row[key]}"')
+    # Test to see if C values are outside of 0..1
+    def update_out_of_range(any_out_of_range, block):
+        if numpy.any(block[numpy.any([block < 0, block > 1])]):
```

Comment:
I think this does work, but the logic could be simplified:
```suggestion
        if numpy.any((block < 0) | (block > 1)):
```

Reply:
Or even further:
```suggestion
        return any_out_of_range or numpy.any((block < 0) | (block > 1))
```
which has the neat advantage of using short-circuit evaluation: if `any_out_of_range` is already `True`, everything after the `or` doesn't need to be evaluated at all!